cmd/tempo-cli/cmd-gen-index.go

```diff
@@ -53,7 +54,7 @@ func ReplayBlockAndGetRecords(meta *backend.BlockMeta, filepath string) ([]commo
 			break
 		}
 
-		iter := encoding.NewIterator(bytes.NewReader(buffer), objectRW)
+		iter := v2.NewIterator(bytes.NewReader(buffer), objectRW)
```

Comment:
can the concept of an iterator even exist with the parquet formats? cli is going to require some real thought as we move to parquet. a lot of these commands won't even make sense anymore and we'll probably want entirely new ones.

easiest thing may be to just drop all cli options that no longer work with parquet, cut tempo 2.0 to justify the breaking changes, and then dust our hands off convincingly.

Reply:
Yes we could make an iterator for `tempopb.Trace`. It would work for every block type, but much slower too.  It wouldn't be used for any normal functions, but it is somewhat useful as a CLI command to dump the contents of a block.  This specific command `gen index` isn't relevant to parquet at all.

I don't think we _have_ to drop the cli commands.  One possibility to leave the existing commands working against a proto (v2) block but error out on a parquet block.